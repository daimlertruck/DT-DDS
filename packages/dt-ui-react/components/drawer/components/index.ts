export * from './title';
export * from './header';
export * from './body';
export * from './scrollEffectContainer';
