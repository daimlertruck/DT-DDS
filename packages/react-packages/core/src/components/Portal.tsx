import React from 'react';
import { createPortal } from 'react-dom';

interface PortalProps {
  children: React.ReactNode;
  isOpen: boolean;
}

/**
 * Portals provide a first-class way to render children into a
 * DOM node that exists outside the DOM hierarchy of the parent component.
 * reference -> https://reactjs.org/docs/portals.html
 */
export const Portal = ({ children, isOpen }: PortalProps) => {
  return isOpen ? createPortal(children, document.body) : null;
};
