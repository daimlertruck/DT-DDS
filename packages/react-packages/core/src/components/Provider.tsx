import { ThemeProvider, Global } from '@emotion/react';
import React from 'react';

import { theme as defaultTheme, globalStyles } from '../themes';
import { FontFace, CustomTheme as Theme } from '../types';

type ThemeProviderType = {
  theme?: Theme;
  webFonts?: FontFace[];
  children: React.ReactNode;
};

// React ThemeProvider wrapper
export const Provider = ({
  theme = defaultTheme,
  webFonts = [],
  children,
}: ThemeProviderType) => {
  return (
    <>
      <Global styles={[globalStyles(theme), ...webFonts]} />
      <ThemeProvider theme={theme}>{children}</ThemeProvider>
    </>
  );
};
