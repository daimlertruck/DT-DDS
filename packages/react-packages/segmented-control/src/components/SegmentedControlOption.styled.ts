import styled from '@emotion/styled';

export const SegmentedControlOptionStyled = styled.button<{
  showLabel: boolean;
}>`
  ${({ theme, showLabel }) => `
    ${theme.fontStyles.body2};

    display: flex;
    align-items: center;
    gap: ${theme.spacing['5xs']};
    cursor: pointer;
    background-color: ${theme.palette.base.default};
    color: ${theme.palette.content.body};
    border: none;
    border-radius: ${theme.shape.button};
    padding: ${theme.spacing['5xs']} ${
    theme.spacing[showLabel ? '3xs' : '5xs']
  };

    &:not(:disabled):hover {
      background-color: ${theme.palette.base.light};
      color: ${theme.palette.content.heading};

      i {
        color: ${theme.palette.content.heading};
      }
    }

    &:not(:disabled)[aria-pressed="true"] {
      background-color: ${theme.palette.primary.default};
      color: ${theme.palette.content.contrast};

      i {
        color: ${theme.palette.content.contrast};
      }
    }

    &:disabled {
        color: ${theme.palette.content.disabled};
        cursor: not-allowed;

        i {
          color: ${theme.palette.content.disabled};
        }
    }

  `}
`;
