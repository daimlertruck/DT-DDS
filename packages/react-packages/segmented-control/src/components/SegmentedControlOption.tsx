import { Code } from '@dt-ui/icons';
import { BaseProps } from '@dt-ui/react-core';
import { Icon } from '@dt-ui/react-icon';
import { Tooltip } from '@dt-ui/react-tooltip';

import { useSegmentedControlContext } from '../context';

import { SegmentedControlOptionStyled } from './SegmentedControlOption.styled';

export interface SegmentedControlProps extends BaseProps {
  label: string;
  value: string;
  iconCode?: Code;
  disabled?: boolean;
}

export const SegmentedControlOption = ({
  label,
  iconCode,
  value,
  disabled,
}: SegmentedControlProps) => {
  const { setSelectedValue, showLabel, selectedValue } =
    useSegmentedControlContext();

  const isSelected = selectedValue === value;
  const isLabelVisible = showLabel || !iconCode;

  const renderOption = (
    <SegmentedControlOptionStyled
      aria-label={label}
      aria-pressed={isSelected ? 'true' : 'false'}
      disabled={disabled}
      onClick={() => setSelectedValue(value)}
      showLabel={isLabelVisible}
      type='button'
    >
      {iconCode ? <Icon code={iconCode} size='medium' /> : null}
      {isLabelVisible ? label : null}
    </SegmentedControlOptionStyled>
  );

  return isLabelVisible ? (
    renderOption
  ) : (
    <Tooltip>
      {renderOption}
      <Tooltip.Content>{label}</Tooltip.Content>
    </Tooltip>
  );
};
