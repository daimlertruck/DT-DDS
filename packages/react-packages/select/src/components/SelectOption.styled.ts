import styled from '@emotion/styled';

interface SelectOptionProps {
  isHighlighted: boolean;
  isMulti: boolean;
}

interface CheckIconWrapperProps {
  isCheckHidden: boolean;
}

export const SelectOptionStyled = styled.li<SelectOptionProps>`
  display: flex;

  ${({ theme, isMulti, isHighlighted }) => `
    cursor:pointer;
    align-items: center;
    ${!isMulti && `padding: ${theme.spacing['4xs']} ${theme.spacing['2xs']}`};
    gap: ${isMulti ? theme.spacing['4xs'] : theme.spacing['5xs']};
    background-color: ${
      isHighlighted && !isMulti
        ? theme.palette.primary.light
        : theme.palette.base.default
    };
    color: ${theme.palette.content.body};
    ${theme.fontStyles.body2}

    & > label {
      ${isMulti && `padding: ${theme.spacing['4xs']} ${theme.spacing['2xs']}`};
    }
    
    &:hover,&:focus-within {
      background-color: ${theme.palette.primary.light};
    }

    &[aria-disabled="true"] {
      cursor: not-allowed;
      color: ${theme.palette.content.disabled};
      &:hover {
        background-color: ${theme.palette.base.default};
      }
    }
  `}
`;

export const SelectOptionContentStyled = styled.div`
  text-overflow: ellipsis;
  white-space: nowrap;
  overflow: hidden;
  width: 100%;
`;

export const CheckIconWrapperStyled = styled.div<CheckIconWrapperProps>`
  display: flex;

  ${({ isCheckHidden }) => isCheckHidden && 'visibility: hidden;'};
`;
