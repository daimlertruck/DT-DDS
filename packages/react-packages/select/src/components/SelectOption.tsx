import { Checkbox } from '@dt-dds/react-checkbox';
import { BaseProps } from '@dt-dds/react-core';

import { useSelectContext } from '../context';

import {
  SelectOptionStyled,
  SelectOptionContentStyled,
} from './SelectOption.styled';

export interface SelectOptionProps extends BaseProps {
  value: string;
  label?: string;
  index: number;
  disabled?: boolean;
}

export const SelectOption = ({
  dataTestId,
  index,
  children,
  style,
  value,
  label,
  disabled,
}: SelectOptionProps) => {
  const { getItemProps, isItemHighlighted, isItemSelected, isMulti } =
    useSelectContext();

  const item = { value: value, label: label, disabled: !!disabled };

  const isSelected = isItemSelected(item);

  const isHighlighted = isItemHighlighted(index);
  const itemProps = getItemProps({ item, index });

  return (
    <SelectOptionStyled
      aria-disabled={itemProps['aria-disabled']}
      data-testid={dataTestId}
      isHighlighted={isHighlighted}
      isMulti={isMulti}
      style={style}
      {...(!isMulti && {
        ...itemProps,
        'aria-selected': isSelected,
      })}
    >
      {isMulti ? (
        <Checkbox
          isChecked={isSelected}
          isDisabled={itemProps['aria-disabled']}
          {...itemProps}
        >
          {children || label || value}
        </Checkbox>
      ) : (
        <SelectOptionContentStyled isSelected={isSelected}>
          {children || label || value}
        </SelectOptionContentStyled>
      )}
    </SelectOptionStyled>
  );
};
