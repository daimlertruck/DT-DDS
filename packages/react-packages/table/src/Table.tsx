import { BaseProps } from '@dt-ui/react-core';
import { useEffect, useRef, useState } from 'react';

import {
  TableBody,
  TableColumnHeader,
  TableDataCell,
  TableHead,
  TableRow,
} from './components';
import { TableContextProvider } from './context';
import { TableStyled } from './Table.styled';
import { observeShouldColumnsBeFixed } from './utils/fixedColumns';

export interface TableProps extends BaseProps {
  isFixed?: boolean;
  fixedColumnCount?: number;
  fixedEndColumnCount?: number;
}

const Table = ({
  children,
  dataTestId,
  style,
  isFixed = false,
  fixedColumnCount = 0,
  fixedEndColumnCount = 0,
}: TableProps) => {
  const tableRef = useRef<HTMLTableElement>(null);

  const [isColumnsFixed, setIsColumnsFixed] = useState(false);

  useEffect(() => {
    const observer = observeShouldColumnsBeFixed(
      tableRef,
      fixedColumnCount,
      fixedEndColumnCount,
      setIsColumnsFixed
    );
    return () => observer.disconnect();
  }, [fixedColumnCount, fixedEndColumnCount]);

  return (
    <TableContextProvider
      fixedColumnCount={fixedColumnCount}
      fixedEndColumnCount={fixedEndColumnCount}
      isColumnsFixed={isColumnsFixed}
    >
      <TableStyled
        data-testid={dataTestId ?? 'table'}
        isFixed={isFixed}
        ref={tableRef}
        style={style}
      >
        {children}
      </TableStyled>
    </TableContextProvider>
  );
};

Table.Head = TableHead;
Table.ColumnHeader = TableColumnHeader;
Table.Body = TableBody;
Table.Row = TableRow;
Table.DataCell = TableDataCell;

export default Table;
