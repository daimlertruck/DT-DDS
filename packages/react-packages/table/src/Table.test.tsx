import { withProviders } from '@dt-ui/react-core';
import { render, screen, within, waitFor } from '@testing-library/react';
import '@emotion/jest';

import { TableHeadProps, TableDataCellProps } from './components';
import { default as Table, TableProps } from './Table';

const COLUMNS = ['Column 1', 'Column 2', 'Column 3', 'Column 4'];
const ROWS = [
  [
    'Row 1 - content 1',
    'Row 1 - content 2',
    'Row 1 - content 3',
    'Row 1 - content 4',
  ],
  [
    'Row 2 - content 1',
    'Row 2 - content 2',
    'Row 2 - content 3',
    'Row 2 - content 4',
  ],
  [
    'Row 3 - content 1',
    'Row 3 - content 2',
    'Row 3 - content 3',
    'Row 3 - content 4',
  ],
];

const ProvidedTable = withProviders(Table);

const renderTableWithProps = (
  {
    isFixed,
    hasFixedHeader,
    textAlign,
    columnWidth,
  }: TableProps & TableHeadProps & TableDataCellProps = {
    isFixed: false,
    hasFixedHeader: false,
    textAlign: 'left',
    columnWidth: '',
  }
) => {
  const fixedColumnCount = 1;
  const fixedEndColumnCount = 2;

  return render(
    <ProvidedTable
      fixedColumnCount={fixedColumnCount}
      fixedEndColumnCount={fixedEndColumnCount}
      isFixed={isFixed}
    >
      <Table.Head hasFixedHeader={hasFixedHeader}>
        <Table.Row>
          {COLUMNS.map((column: string) => (
            <Table.ColumnHeader
              key={`column-header-${column}`}
              textAlign={textAlign}
            >
              {column}
            </Table.ColumnHeader>
          ))}
        </Table.Row>
      </Table.Head>
      <Table.Body>
        {ROWS.map((row: string[]) => (
          <Table.Row key={row.toString()} onClick={jest.fn()}>
            {row.map((content: string, i: number) => (
              <Table.DataCell
                columnWidth={columnWidth}
                key={`column-${COLUMNS[i]}-${content}`}
                textAlign={textAlign}
              >
                {content}
              </Table.DataCell>
            ))}
          </Table.Row>
        ))}
      </Table.Body>
    </ProvidedTable>
  );
};

describe('<Table /> component', () => {
  let mockIntersectionObserve: jest.SpyInstance;

  beforeEach(() => {
    mockIntersectionObserve = jest.spyOn(window, 'IntersectionObserver');
    mockIntersectionObserve.mockImplementationOnce((callback) => {
      callback([{ isIntersecting: false }]);
      return { observe: jest.fn(), disconnect: jest.fn() };
    });
  });

  it('renders a Basic Table', () => {
    const { container } = renderTableWithProps({ isFixed: true });

    expect(container).toMatchSnapshot();
  });

  it('renders a Basic Table without fixed columns', () => {
    const { container } = render(
      <ProvidedTable>
        <Table.Head>
          <Table.Row>
            {COLUMNS.map((column: string) => (
              <Table.ColumnHeader
                key={`column-header-${column}`}
                textAlign='left'
              >
                {column}
              </Table.ColumnHeader>
            ))}
          </Table.Row>
        </Table.Head>
        <Table.Body>
          {ROWS.map((row: string[]) => (
            <Table.Row key={row.toString()}>
              {row.map((content: string, i: number) => (
                <Table.DataCell
                  columnWidth=''
                  key={`column-${COLUMNS[i]}-${content}`}
                  textAlign='left'
                >
                  {content}
                </Table.DataCell>
              ))}
            </Table.Row>
          ))}
        </Table.Body>
      </ProvidedTable>
    );

    expect(container).toMatchSnapshot();
  });

  it('renders a Composable Table with a custom data structure', () => {
    renderTableWithProps();

    // Test head columns are rendered
    COLUMNS.forEach((column: string) => {
      const cell = within(screen.getByTestId('table-head'))
        .getByText(column)
        .closest('th');

      const utils = within(cell!);

      expect(utils.getByText(column)).toBeInTheDocument();
    });

    // Test rows content are in rendered
    ROWS.forEach((row: string[]) => {
      row.forEach((cell: string) => {
        const dataCell = screen.getByText(cell).closest('td');
        const utils = within(dataCell!);

        expect(utils.getByText(cell)).toBeInTheDocument();
      });
    });
  });

  it('applies the expected styles when Table has Fixed Header', () => {
    renderTableWithProps({ hasFixedHeader: true });

    const theadElement = screen.getByTestId('table-head');

    expect(theadElement).toHaveStyleRule('position', 'sticky');
  });

  it('applies custom column Width to table data cell', () => {
    renderTableWithProps({ columnWidth: '20px' });

    const tdElements = screen.getAllByTestId('table-data-cell');

    tdElements.map(async (e) => {
      await waitFor(() => expect(e).toHaveStyleRule('max-width', '20px'));
    });
  });

  it('applies the expected styles when Table has text aligned center', () => {
    renderTableWithProps({ textAlign: 'center' });

    const thElements = screen.getAllByTestId('table-column-header');
    const tdElements = screen.getAllByTestId('table-data-cell');

    expect(thElements).toHaveLength(4);
    thElements.map((e) => {
      expect(e).toHaveStyleRule('text-align', 'center');
    });

    tdElements.map((e) => {
      expect(e).toHaveStyleRule('text-align', 'center');
    });
  });

  it('applies the expected styles when Table has fixed columns', () => {
    renderTableWithProps();

    const fixedColumn = screen.getByText('Column 1');
    const fixedEndColumn1 = screen.getByText('Column 3');
    const fixedEndColumn2 = screen.getByText('Column 4');

    expect(fixedColumn).toHaveStyleRule('position', 'sticky');
    expect(fixedEndColumn1).toHaveStyleRule('position', 'sticky');
    expect(fixedEndColumn2).toHaveStyleRule('position', 'sticky');
  });
});
