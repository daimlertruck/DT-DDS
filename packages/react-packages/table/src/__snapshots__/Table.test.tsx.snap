// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Table /> component renders a Basic Table 1`] = `
.emotion-0 {
  table-layout: fixed;
  width: 100%;
  border-spacing: 0;
}

.emotion-2 {
  display: table-header-group;
}

.emotion-4 {
  display: table-row;
  border-radius: 0px;
  box-shadow: none;
  padding: 0;
  width: 100%;
  background-color: #FFFFFF;
  cursor: auto;
}

.emotion-6 {
  text-align: left;
  color: #292929;
  padding: 0 16px 16px;
  font-weight: 700;
  font-size: 0.75rem;
  line-height: 1rem;
  letter-spacing: normal;
  margin: 0;
  text-transform: none;
  left: 0px;
  right: unset;
  position: -webkit-sticky;
  position: sticky;
  z-index: 2;
  background: #FFFFFF;
  box-shadow: 1px 0 0 0 #F3F3F5;
}

.emotion-8 {
  text-align: left;
  color: #292929;
  padding: 0 16px 16px;
  font-weight: 700;
  font-size: 0.75rem;
  line-height: 1rem;
  letter-spacing: normal;
  margin: 0;
  text-transform: none;
}

.emotion-10 {
  text-align: left;
  color: #292929;
  padding: 0 16px 16px;
  font-weight: 700;
  font-size: 0.75rem;
  line-height: 1rem;
  letter-spacing: normal;
  margin: 0;
  text-transform: none;
  left: unset;
  right: 0px;
  position: -webkit-sticky;
  position: sticky;
  z-index: 2;
  background: #FFFFFF;
  box-shadow: -1px 0 0 0 #F3F3F5;
}

.emotion-12 {
  text-align: left;
  color: #292929;
  padding: 0 16px 16px;
  font-weight: 700;
  font-size: 0.75rem;
  line-height: 1rem;
  letter-spacing: normal;
  margin: 0;
  text-transform: none;
  left: unset;
  right: 0px;
  position: -webkit-sticky;
  position: sticky;
  z-index: 2;
  background: #FFFFFF;
  box-shadow: unset;
}

.emotion-14 {
  display: table-header-group;
}

.emotion-14 tr:hover td {
  background-color: #FAFAFA;
}

.emotion-16 {
  display: table-row;
  border-radius: 0px;
  box-shadow: none;
  padding: 0;
  width: 100%;
  background-color: #FFFFFF;
  cursor: pointer;
}

.emotion-18 {
  display: table-cell;
  min-width: 110px;
  word-break: break-word;
  border-top: 1px solid #F3F3F5;
  padding: 16px;
  text-align: left;
  color: #454545;
  font-weight: 400;
  font-size: 0.875rem;
  line-height: 1.25rem;
  letter-spacing: normal;
  margin: 0;
  text-transform: none;
  left: 0px;
  right: unset;
  position: -webkit-sticky;
  position: sticky;
  z-index: 1;
  background: #FFFFFF;
  box-shadow: 1px 0 0 0 #F3F3F5;
}

.emotion-18:first-of-type {
  padding-top: 16px;
  border-bottom: none;
}

.emotion-20 {
  display: table-cell;
  min-width: 110px;
  word-break: break-word;
  border-top: 1px solid #F3F3F5;
  padding: 16px;
  text-align: left;
  color: #454545;
  font-weight: 400;
  font-size: 0.875rem;
  line-height: 1.25rem;
  letter-spacing: normal;
  margin: 0;
  text-transform: none;
}

.emotion-20:first-of-type {
  padding-top: 16px;
  border-bottom: none;
}

.emotion-22 {
  display: table-cell;
  min-width: 110px;
  word-break: break-word;
  border-top: 1px solid #F3F3F5;
  padding: 16px;
  text-align: left;
  color: #454545;
  font-weight: 400;
  font-size: 0.875rem;
  line-height: 1.25rem;
  letter-spacing: normal;
  margin: 0;
  text-transform: none;
  left: unset;
  right: 0px;
  position: -webkit-sticky;
  position: sticky;
  z-index: 1;
  background: #FFFFFF;
  box-shadow: -1px 0 0 0 #F3F3F5;
}

.emotion-22:first-of-type {
  padding-top: 16px;
  border-bottom: none;
}

.emotion-24 {
  display: table-cell;
  min-width: 110px;
  word-break: break-word;
  border-top: 1px solid #F3F3F5;
  padding: 16px;
  text-align: left;
  color: #454545;
  font-weight: 400;
  font-size: 0.875rem;
  line-height: 1.25rem;
  letter-spacing: normal;
  margin: 0;
  text-transform: none;
  left: unset;
  right: 0px;
  position: -webkit-sticky;
  position: sticky;
  z-index: 1;
  background: #FFFFFF;
  box-shadow: unset;
}

.emotion-24:first-of-type {
  padding-top: 16px;
  border-bottom: none;
}

<div>
  <table
    class="emotion-0 emotion-1"
    data-testid="table"
  >
    <thead
      class="emotion-2 emotion-3"
      data-testid="table-head"
    >
      <tr
        class="emotion-4 emotion-5"
        data-testid="table-row"
      >
        <th
          class="emotion-6 emotion-7"
          data-column-index="0"
          data-fixed-position="0"
          data-testid="table-column-header"
        >
          Column 1
        </th>
        <th
          class="emotion-8 emotion-7"
          data-testid="table-column-header"
        >
          Column 2
        </th>
        <th
          class="emotion-10 emotion-7"
          data-column-index="2"
          data-fixed-position="0"
          data-testid="table-column-header"
        >
          Column 3
        </th>
        <th
          class="emotion-12 emotion-7"
          data-column-index="3"
          data-fixed-position="0"
          data-testid="table-column-header"
        >
          Column 4
        </th>
      </tr>
    </thead>
    <tbody
      class="emotion-14 emotion-15"
      data-testid="table-body"
    >
      <tr
        class="emotion-16 emotion-5"
        data-testid="table-row"
      >
        <td
          class="emotion-18 emotion-19"
          data-column-index="0"
          data-fixed-position="0"
          data-testid="table-data-cell"
        >
          Row 1 - content 1
        </td>
        <td
          class="emotion-20 emotion-19"
          data-testid="table-data-cell"
        >
          Row 1 - content 2
        </td>
        <td
          class="emotion-22 emotion-19"
          data-column-index="2"
          data-fixed-position="0"
          data-testid="table-data-cell"
        >
          Row 1 - content 3
        </td>
        <td
          class="emotion-24 emotion-19"
          data-column-index="3"
          data-fixed-position="0"
          data-testid="table-data-cell"
        >
          Row 1 - content 4
        </td>
      </tr>
      <tr
        class="emotion-16 emotion-5"
        data-testid="table-row"
      >
        <td
          class="emotion-18 emotion-19"
          data-column-index="0"
          data-fixed-position="0"
          data-testid="table-data-cell"
        >
          Row 2 - content 1
        </td>
        <td
          class="emotion-20 emotion-19"
          data-testid="table-data-cell"
        >
          Row 2 - content 2
        </td>
        <td
          class="emotion-22 emotion-19"
          data-column-index="2"
          data-fixed-position="0"
          data-testid="table-data-cell"
        >
          Row 2 - content 3
        </td>
        <td
          class="emotion-24 emotion-19"
          data-column-index="3"
          data-fixed-position="0"
          data-testid="table-data-cell"
        >
          Row 2 - content 4
        </td>
      </tr>
      <tr
        class="emotion-16 emotion-5"
        data-testid="table-row"
      >
        <td
          class="emotion-18 emotion-19"
          data-column-index="0"
          data-fixed-position="0"
          data-testid="table-data-cell"
        >
          Row 3 - content 1
        </td>
        <td
          class="emotion-20 emotion-19"
          data-testid="table-data-cell"
        >
          Row 3 - content 2
        </td>
        <td
          class="emotion-22 emotion-19"
          data-column-index="2"
          data-fixed-position="0"
          data-testid="table-data-cell"
        >
          Row 3 - content 3
        </td>
        <td
          class="emotion-24 emotion-19"
          data-column-index="3"
          data-fixed-position="0"
          data-testid="table-data-cell"
        >
          Row 3 - content 4
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`<Table /> component renders a Basic Table without fixed columns 1`] = `
.emotion-0 {
  table-layout: auto;
  width: 100%;
  border-spacing: 0;
}

.emotion-2 {
  display: table-header-group;
}

.emotion-4 {
  display: table-row;
  border-radius: 0px;
  box-shadow: none;
  padding: 0;
  width: 100%;
  background-color: #FFFFFF;
  cursor: auto;
}

.emotion-6 {
  text-align: left;
  color: #292929;
  padding: 0 16px 16px;
  font-weight: 700;
  font-size: 0.75rem;
  line-height: 1rem;
  letter-spacing: normal;
  margin: 0;
  text-transform: none;
}

.emotion-14 {
  display: table-header-group;
}

.emotion-14 tr:hover td {
  background-color: #FAFAFA;
}

.emotion-18 {
  display: table-cell;
  min-width: 110px;
  word-break: break-word;
  border-top: 1px solid #F3F3F5;
  padding: 16px;
  text-align: left;
  color: #454545;
  font-weight: 400;
  font-size: 0.875rem;
  line-height: 1.25rem;
  letter-spacing: normal;
  margin: 0;
  text-transform: none;
}

.emotion-18:first-of-type {
  padding-top: 16px;
  border-bottom: none;
}

<div>
  <table
    class="emotion-0 emotion-1"
    data-testid="table"
  >
    <thead
      class="emotion-2 emotion-3"
      data-testid="table-head"
    >
      <tr
        class="emotion-4 emotion-5"
        data-testid="table-row"
      >
        <th
          class="emotion-6 emotion-7"
          data-testid="table-column-header"
        >
          Column 1
        </th>
        <th
          class="emotion-6 emotion-7"
          data-testid="table-column-header"
        >
          Column 2
        </th>
        <th
          class="emotion-6 emotion-7"
          data-testid="table-column-header"
        >
          Column 3
        </th>
        <th
          class="emotion-6 emotion-7"
          data-testid="table-column-header"
        >
          Column 4
        </th>
      </tr>
    </thead>
    <tbody
      class="emotion-14 emotion-15"
      data-testid="table-body"
    >
      <tr
        class="emotion-4 emotion-5"
        data-testid="table-row"
      >
        <td
          class="emotion-18 emotion-19"
          data-testid="table-data-cell"
        >
          Row 1 - content 1
        </td>
        <td
          class="emotion-18 emotion-19"
          data-testid="table-data-cell"
        >
          Row 1 - content 2
        </td>
        <td
          class="emotion-18 emotion-19"
          data-testid="table-data-cell"
        >
          Row 1 - content 3
        </td>
        <td
          class="emotion-18 emotion-19"
          data-testid="table-data-cell"
        >
          Row 1 - content 4
        </td>
      </tr>
      <tr
        class="emotion-4 emotion-5"
        data-testid="table-row"
      >
        <td
          class="emotion-18 emotion-19"
          data-testid="table-data-cell"
        >
          Row 2 - content 1
        </td>
        <td
          class="emotion-18 emotion-19"
          data-testid="table-data-cell"
        >
          Row 2 - content 2
        </td>
        <td
          class="emotion-18 emotion-19"
          data-testid="table-data-cell"
        >
          Row 2 - content 3
        </td>
        <td
          class="emotion-18 emotion-19"
          data-testid="table-data-cell"
        >
          Row 2 - content 4
        </td>
      </tr>
      <tr
        class="emotion-4 emotion-5"
        data-testid="table-row"
      >
        <td
          class="emotion-18 emotion-19"
          data-testid="table-data-cell"
        >
          Row 3 - content 1
        </td>
        <td
          class="emotion-18 emotion-19"
          data-testid="table-data-cell"
        >
          Row 3 - content 2
        </td>
        <td
          class="emotion-18 emotion-19"
          data-testid="table-data-cell"
        >
          Row 3 - content 3
        </td>
        <td
          class="emotion-18 emotion-19"
          data-testid="table-data-cell"
        >
          Row 3 - content 4
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
